### PR TITLE
storage: Move some fstab/crypttab to utils.js

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
Another PR that just moves code, to get the coverage noise out of the main PR (#19388).

**Review by martinpitt — APPROVED** [review]

File: pkg/storaged/nfs-details.jsx [hunk 1]
```diff
@@ -80,10 +80,10 @@ function get_exported_directories(server) {
 
 export function nfs_fstab_dialog(client, entry) {
     const mount_options = entry ? entry.fields[3] : "defaults";
-    const split_options = format.parse_options(mount_options == "defaults" ? "" : mount_options);
-    const opt_auto = !format.extract_option(split_options, "noauto");
-    const opt_ro = format.extract_option(split_options, "ro");
-    const extra_options = format.unparse_options(split_options);
+    const split_options = parse_options(mount_options == "defaults" ? "" : mount_options);
```
Comment (cockpituous) [inline pkg/storaged/nfs-details.jsx:83]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19411-20230928-074153-7c55399d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 2]
```diff
@@ -47,6 +47,88 @@ export function compare_versions(a, b) {
...
+    }
+}
+
+export function edit_crypto_config(block, modify) {
+    let old_config, new_config;
+
+    function commit() {
+        new_config[1]["track-parents"] = { t: 'b', v: true };
+        if (old_config)
+            return block.UpdateConfigurationItem(old_config, new_config, { });
```
Comment (cockpituous) [inline pkg/storaged/utils.js:79]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19411-20230928-074153-7c55399d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 3]
```diff
@@ -47,6 +47,88 @@ export function compare_versions(a, b) {
...
+        else
+            return block.AddConfigurationItem(new_config, { });
```
Comment (cockpituous) [inline pkg/storaged/utils.js:81]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19411-20230928-074153-7c55399d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 4]
```diff
@@ -47,6 +47,88 @@ export function compare_versions(a, b) {
...
+    }
+
+    return block.GetSecretConfiguration({}).then(
+        function (items) {
+            old_config = items.find(c => c[0] == "crypttab");
+            new_config = ["crypttab", old_config ? Object.assign({ }, old_config[1]) : { }];
```
Comment (cockpituous) [inline pkg/storaged/utils.js:87]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19411-20230928-074153-7c55399d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 5]
```diff
@@ -47,6 +47,88 @@ export function compare_versions(a, b) {
...
+            if (!new_config[1]['passphrase-contents'])
+                new_config[1]['passphrase-contents'] = { t: 'ay', v: encode_filename("") };
+
+            return modify(new_config[1], commit);
+        });
+}
+
+export function set_crypto_options(block, readonly, auto, nofail, netdev) {
+    return edit_crypto_config(block, (config, commit) => {
+        const opts = config.options ? parse_options(decode_filename(config.options.v)) : [];
```
Comment (cockpituous) [inline pkg/storaged/utils.js:102]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19411-20230928-074153-7c55399d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
TF seems stuck in Copr.

**Pull request merged**